storage/src/main/java/tech/pegasys/artemis/storage/Store.java

```diff
@@ -100,43 +101,6 @@ public Transaction startTransaction() {
     return new Transaction();
   }
 
-  public Transaction startTransaction(UnsignedLong slot) {
-    return new Transaction(slot);
-  }
-
-  public void cleanStoreUntilSlot(UnsignedLong slot) {
```

Comment:
Why cut this cleanup code?  Seems like we could modify this a bit and run it when a new block is finalized.

Reply:
Agreed

Reply:
I'm pretty ruthless about deleting unused code as soon as it requires any changes to keep it "working" because it can always be pulled back out of git history if needed later.

But in this case the code is walking through all the in-memory blocks & states to prune which is inefficient and the disk storage has already got the optimised approach to this implemented so it can just send an event with the list of block roots to prune as part of the confirmation that the changes have been stored to disk, allowing the in-memory store to be pruned very efficiently.